src/components/tools/Charts/DotAndWhisker/Chart.svelte

```diff
@@ -89,33 +80,63 @@
       .align(0);
 
     // Update Y Domain
-    ymin = min(data, (d) => d.lowerci);
-    ymax = max(data, (d) => d.upperci);
-    if (yAxis.baseValue === 0) {
-      ymin = yAxis.baseValue;
+    if (typeof yAxis.domainMin === "number" && !isNaN(yAxis.domainMin)) {
+      ymin = yAxis.domainMin;
+    } else {
+      ymin = min(data, (d) => d.ci_lower);
+    }
+
+    if (typeof yAxis.domainMax === "number" && !isNaN(yAxis.domainMax)) {
+      ymax = yAxis.domainMax;
+    } else {
+      ymax = max(data, (d) => d.ci_upper);
+    }
+
+    if (
+      typeof yAxis.niceMax === "number" &&
+      !isNaN(yAxis.niceMax) &&
+      yAxis.niceMax > ymax
+    ) {
+      ymax = yAxis.niceMax;
     }
 
     // Update Legend
     legendItems.set(
-      xKeys.map((key) => {
-        const series = data.find((d) => d.key === key);
+      xKeys.map((id) => {
+        const series = data.find((d) => d.id === id);
         return {
-          key: series.key,
+          id: series.id,
           label: series.label,
           color: series.color,
           visible: true,
         };
       })
     );
-    console.log($legendItems);
     setContext("Legend", legendItems);
   }
+
+  function createTooltip(d) {
```

Comment:
Since the ToolTip component is local to this Chart component only, why not pass in the props to the component and render the HTML there using Svelte's template syntax vs. doing string concatenation here which is more difficult to read / follow?

Reply:
Good idea. Implemented in https://github.com/berkeley-gif/caladapt-website-2021/pull/163/commits/ddcd93d89503c24014abce04eaa7ce0370831030